tools/report-converter/codechecker_report_converter/analyzer_result.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+# -------------------------------------------------------------------------
+#                     The CodeChecker Infrastructure
+#   This file is distributed under the University of Illinois Open Source
+#   License. See LICENSE.TXT for details.
+# -------------------------------------------------------------------------
+
+from __future__ import print_function
+from __future__ import division
+from __future__ import absolute_import
+
+from abc import ABCMeta, abstractmethod
+import io
+import logging
+import os
+import plistlib
+import sys
+
+from .report import generate_report_hash
+
+LOG = logging.getLogger('ReportConverter')
+
+
+class AnalyzerResult(object):
+    """ Base class to transform analyzer result. """
+
+    __metaclass__ = ABCMeta
+
+    TOOL_NAME = None
+
+    def transform(self, analyzer_result, output_dir):
+        """ Creates plist files from the given analyzer result to the given
+        output directory.
+        """
+        analyzer_result = os.path.abspath(analyzer_result)
+        plist_objs = self.parse(analyzer_result)
+
+        self._post_process_result(plist_objs)
+
+        self._write(plist_objs, output_dir)
+
+    @abstractmethod
+    def parse(self, analyzer_result):
+        """ Creates plist objects from the given analyzer result.
+
+        Returns a list of plist objects.
+        """
+        raise NotImplementedError("Subclasses should implement this!")
+
+    def _post_process_result(self, plist_objs):
+        """ Post process the parsed result.
+
+        By default it will add report hashes for the diagnostics.
+        """
+        self._add_report_hash(plist_objs)
+
+    def _add_report_hash(self, plist_objs):
+        """ Generate report hash for each diagnostics in the plist objects. """
+        for plist_data in plist_objs:
+            files = plist_data['files']
+            for diag in plist_data['diagnostics']:
+                report_hash = \
+                    generate_report_hash(diag,
+                                         files[diag['location']['file']])
+                diag['issue_hash_content_of_line_in_context'] = report_hash
+
+    def _get_analyzer_result_file_content(self, result_file):
+        """ Return the content of the given file. """
+        with io.open(result_file, 'r', encoding='utf-8',
+                     errors='replace') as analyzer_result:
+            return analyzer_result.readlines()
+
+    def _write(self, plist_objs, output_dir):
+        """ Creates plist files from the parse result to the given output. """
+        if not plist_objs:
+            LOG.info("No '%s' results can be found in the given code analyzer "
+                     "output.", self.TOOL_NAME)
+            sys.exit(0)
+
+        output_dir = os.path.abspath(output_dir)
+        for plist_data in plist_objs:
+            file_name = os.path.basename(plist_data['files'][0])
+            out_file_name = '{0}_{1}.plist'.format(file_name, self.TOOL_NAME)
```

Comment:
what happens if there are multiple plist_data objects for the same source file and analyzer? Will the second iteration overwrite the first written file? Is it somewhere checked that there can not be two entries in the plist_objs where these values are the same?

Reply:
In the `PlistConverter` we will create a dictionary called `path_to_plist` which will map the report file path to a plist object:
https://github.com/csordasmarton/codechecker/blob/2d3a705027279412168ecda4cbbdb5ada37c1181/tools/report-converter/codechecker_report_converter/plist_converter.py#L72-L74

So if multiple bugs can be found in the same source file `dummy.cpp`, it will create a plist file with multiple diagnostics.